### DISCUSSION TITLE
HashDB Filter Know Hashs

### DISCUSSION THREAD

**Discussion by thiagofuer** [sepinf-inc/IPED#1751] [Q&A] [answered]
Hi folks
I'm analysing my evidence processing result using iped-hashes.db, created using the [nist](https://s3.amazonaws.com/rds.nsrl.nist.gov/RDS/rds_2023.06.1/RDS_2023.06.1_android_minimal.zip) RDS V3 Android Minimal
I've set the 'hashesDB' propertie on LocalConfig.txt 
-hashesDB = /home/thiago/Downloads/nist/android_hash/ipedhashdb/iped-hashes.db
i've enabled the follow properties on IpedConfig.txt 
-enableHash = true
-enableHashDBLookup = true
-ignoreDuplicates = true

so i've process an Android extraction using iped 4.1.2 
when processing i saw some ignored itens
 ![image](https://github.com/sepinf-inc/IPED/assets/1029909/f3583f0f-02c9-44c3-b635-feb90a520c12)
and got 90.337 files as restult:
![image](https://github.com/sepinf-inc/IPED/assets/1029909/604cb6bd-8211-47a1-8e6a-66f22d6393eb)
I tryed to filter my results to show only the know hash results and i got the same total files
![image](https://github.com/sepinf-inc/IPED/assets/1029909/d83e11bd-62d0-4b14-9b71-753092f2322b)

The question is: How can i see the "Ignored itens"? or how to filter the know hash files?

So i did the same processing using iped 4.0.1
I saw less files (80.911), but i can understando because this process has diferent enabled/disabled parsers 
![image](https://github.com/sepinf-inc/IPED/assets/1029909/4def738d-185d-478d-81e9-067c62501a02)
After use the filter "know hash results" i got 79.871 files..
![image](https://github.com/sepinf-inc/IPED/assets/1029909/31d10758-41b8-4598-bcda-1cc88c3e4011)
and now I understood that the files that have the known hash were filtered out
but now another question has arisen...
 the filtered file dont appears to be a android know hashdb file..
![image](https://github.com/sepinf-inc/IPED/assets/1029909/460787a6-972b-47b3-a14e-2e31d558f958)
So please could anyone help me to understood how to filter the know hashDB?

**Comment by lfcnassif** [comment, marked as answer]
Hi @thiagofuer,

The "Filter Known Hashes"/"Filtrar Hashes Conhecidos" filter removes from result set the known files. If you want the opposite behavior (list known files), you can create a new filter negating the expression of the default one, take a look at "Manage Filters" menu entry in the Options button at top-right.

About the same number of results got after applying the filter in the first case, probably the "excludeKnown" option in conf/HashDBLookupConfig.txt was set to true in the used profile. Probably it was set to false (default value) in the processing using version 4.0.1.

About the last screenshot, as I said, the listed files are not known, the known files were filtered out (as you also said).

**Reply by thiagofuer** [reply]
thanks for the reply, i will try it again

**Reply by lfcnassif** [reply]
As a side note, please see the `ignoreDuplicates` option is very dangerous, you can miss many email attachments, that is warned in IPEDConfig.txt. I considered the removal of that option in the past but didn't after some users asked to keep the option, regardless of its risks...

**Comment by thiagofuer** [comment]
I've disabled the ignoreDuplicates and excludeKnown and now i can see the know files
![image](https://github.com/sepinf-inc/IPED/assets/1029909/97a33044-42c1-4312-93f2-5463a706d476)
Another question... whats the sintax used on filters?

**Reply by lfcnassif** [reply]
The same used by the search bar (Lucene).

**Reply by thiagofuer** [reply]
thanks a lot
